app/scripts/lib/app-start.js

```diff
@@ -482,18 +486,26 @@ function (
 
     /**
      * Sets a UUID value that is unrelated to any account information.
-     * This value is useful to determine if the logged out user qualifies for ab testing or metrics.
+     * This value is useful to determine if the logged out user qualifies
+     * for A/B testing or metrics.
      */
     initializeUuid: function () {
       var storage = Storage.factory('localStorage', this._window);
 
-      // we reuse the same uuid if it is available in local storage.
-      if (storage.get('uuid')) {
+      // Reuse the same uuid if it is available in local storage.
+      if (this._searchParam('resume')) {
+        var resumeParam = this._searchParam('resume');
+        var resumeToken = new ResumeToken(ResumeToken.parse(resumeParam));
```

Comment:
See [this comment](https://github.com/mozilla/fxa-content-server/pull/2688/files#r33918999) about moving the creation logic to a static function on ResumeToken.